File: exercises/week5/series/sources-solution/compute_pi.py
from __future__ import print_function
import math
from series import Series
################################################################


class ComputePi(Series):

    def __init__(self,):
        Series.__init__(self)
        self.current_value = 0

    def __del__(self):
        pass

    def compute(self, N):
        Series.compute(self, N)
        return math.sqrt(6.*self.current_value)

    def computeTerm(self, k):
        return 1./(1.*k*k)

    def getAnalyticPrediction(self):
        return math.pi

File: exercises/week6/series/sources-solution/src/compute_pi.cc
#include "compute_pi.hh"
#include <cmath>
/* -------------------------------------------------------------------------- */

ComputePi::ComputePi() {
  func = [](UInt i) { return 1. / (1. * i * i); };
}

/* -------------------------------------------------------------------------- */

double ComputePi::compute(UInt N) {
  Series::compute(N);
  return sqrt(6. * current_value);
}

/* -------------------------------------------------------------------------- */

double ComputePi::getAnalyticPrediction() { return M_PI; }

/* -------------------------------------------------------------------------- */
